src/silly_sprites/opengl/shader_program.hpp

```diff
@@ -0,0 +1,38 @@
+#pragma once
+
+#include "../types.hpp"
+#include "magic_enum_wrapper.hpp"
+#include <glad/gl.h>
+#include <string_view>
+#include <tl/optional.hpp>
+
+namespace sly::gl {
+    class ShaderProgram final {
+    public:
+        enum class Type : GLenum {
+            Vertex = GL_VERTEX_SHADER,
+            Geometry = GL_GEOMETRY_SHADER,
+            Fragment = GL_FRAGMENT_SHADER,
+        };
+
+    private:
+        using Shader = tl::optional<GLuint>;
+        GLuint m_program_id;
```

Comment:
Should we call those `id` or `name`? The official name is `name`, that's why I'm asking.